### PR TITLE
to  add custom test cases to handle utf-8 strings

### PR DESCRIPTION
**Pull request by kkumar495**
to fix the build error :
```
[ 68%] Linking CXX shared library libWPEFrameworkCOM.so
/usr/bin/ld: cannot find -lWPEFrameworkWarningReporting::WPEFrameworkWarningReporting
collect2: error: ld returned 1 exit status
```
Add custom test cases in json_testparser.cpp to check if thunder is able to handle utf-8 strings

### CODE REVIEW

File: Source/com/Communicator.h [hunk 1]
```diff
@@ -1152,11 +1154,14 @@ namespace RPC {
 
                     // Anounce the interface as completed
                     string jsonDefaultCategories(Trace::TraceUnit::Instance().Defaults());
+#ifdef WARNING_REPORTING
                     string jsonDefaultWarningCategories(WarningReporting::WarningReportingUnit::Instance().Defaults());
+#endif
                     void* result = _parent.Announce(proxyChannel, message->Parameters());
 
+#ifdef WARNING_REPORTING
                     message->Response().Set(instance_cast<void*>(result), proxyChannel->Extension().Id(), _parent.ProxyStubPath(), jsonDefaultCategories, jsonDefaultWarningCategories);
```
Comment (MFransen69) [inline Source/com/Communicator.h]: I just checked and the unintended forced WARNING_REPORTING dependencies here and above were already fixed in the mean time ( that is the reason the PullRequest has merge conflicts) (and just to make sure this line is not only needed when warning reporting is active so it has been corrected in a different way). 
I propose to just remove this and above changes from the pull requests, that will also solve the merge conflicts.

File: Tests/unit/IPTestAdministrator.h [hunk 2]
```diff
@@ -24,6 +24,10 @@
 #include <string>
 #include <time.h>
 
+#ifndef MODULE_NAME
+#define MODULE_NAME Tracing
```
Comment (MFransen69) [inline Tests/unit/IPTestAdministrator.h]: Perhaps for a test not too important but this module name might be confusing :)

**Review by MFransen69 — CHANGES_REQUESTED** [review]

**Comment by kkumar495** [comment]
changes are made  as per the review comments.

**Comment by pwielders** [comment]
@kkumar495 Change set is oke now, but you must sign the CLA before we can merge it!

**Comment by kkumar495** [comment]
> @kkumar495 Change set is oke now, but you must sign the CLA before we can merge it!

Done

File: Source/com/Communicator.h (hunk 1 above)
Comment (kkumar495) [reply Source/com/Communicator.h]: removed this change

**Review by MFransen69 — APPROVED** [review]

**Pull request merged**